File: blacksmith-of-lenai-app/src/app/shared/models/player-skill.ts
import { SkillPerkData } from './skill-perk-data';
import { PlayerSkillExperience } from './player-skill-experience';

export interface PlayerSkill {
  playerSkillId: string;
  skillName: string;
  skillLevel: number;
  currentSkillExp: number;
  skillPerks: SkillPerkData[];
  playerSkillExperience: PlayerSkillExperience;
}

File: blacksmith-of-lenai-api/Models/PlayerSkill.cs
namespace PlayFabApi.Models;

public sealed record PlayerSkill
{
    public required Guid PlayerSkillId { get; init; }
    public required string SkillName { get; init; }
    public required int SkillLevel { get; init; }
    public required int CurrentSkillExp { get; init; }
    public required PlayerSkillExperience PlayerSkillExperience { get; init; }
    public required List<SkillPerk> SkillPerks { get; init; }
}